CharlieBackend.Api.UnitTes/DashboardServiceTest.cs

```diff
@@ -32,12 +54,94 @@ public async Task GetStudentsClassbook()
                 }
             };
 
-            var studentclassbookWrongRequest = new StudentsRequestDto<ClassbookResultType>()
+            _dashboardRepositoryMock.Setup(x => x.GetGroupsIdsByCourseIdAndPeriodAsync(
+                studentclassbookRequestWithData.CourseId.Value,
+                studentclassbookRequestWithData.StartDate,
+                studentclassbookRequestWithData.FinishDate)).ReturnsAsync(new List<long>() { 2 });
+
+            _dashboardRepositoryMock.Setup(x => x.GetStudentsMarksListByGroupIdsAndDate(
+                new List<long> { 2 }, studentclassbookRequestWithData.StartDate, studentclassbookRequestWithData.FinishDate))
+                .ReturnsAsync(new List<StudentMarkDto>() { new StudentMarkDto() { StudentGroupId = 1 } });
+
+            InitializeForDashboardRepository();
+
+            var dashboardService = new DashboardService(
+                unitOfWork: _unitOfWorkMock.Object,
+                currentUserService: _currentUserServiceMock.Object);
+
+            //Act
+            var successResult = await dashboardService.GetStudentsClassbookAsync(studentclassbookRequestWithData);
+
+            //Assert
+            successResult.Data.StudentsMarks.Should().NotBeNull();
```

Comment:
Please verify the properties in the result.

Reply:
Done